### ISSUE TITLE
Add convenience method for qualified name construction

### ISSUE THREAD

**Issue by rtomsick** [jOOQ/jOOQ#6089]
For those of us not using code gen, would it be possible to add a convenience method to `DSL`?

Specifically something like:

```java
public static Name
name(Table<?> tbl, String ... parts)
{
	String[] n = new String[parts.length + 1];
	n[0] = tbl.getName();
	System.arraycopy(parts, 0, n, 1, parts.length);
	return DSL.name(n);
}
```

I don't think this will conflict with any existing part of the API, and for those of us constructing `Name`s manually it will make our code much more readable.

If not, no biggie.  It's easy enough to static import it from wherever one puts their glue code.

**Comment by lukaseder** [comment]
Thank you very much for your request. Incidentally, jOOQ 3.10 already has tons of new `Name` constructing API methods which were all needed for the recent `Parser` improvements (#5955):

- https://github.com/jOOQ/jOOQ/issues/5993 Add Name DSL.quotedName() and unquotedName()
- https://github.com/jOOQ/jOOQ/issues/5998 Add DSL.name(Name...) to construct a Name from Name.last() elements
- https://github.com/jOOQ/jOOQ/issues/6012 Implement specialised Name for unqualified Names
- https://github.com/jOOQ/jOOQ/issues/6013 Add Name Name.qualifier()
- https://github.com/jOOQ/jOOQ/issues/6068 Add Name { Field | Table | Schema | Catalog }.getQualifiedName()
- https://github.com/jOOQ/jOOQ/issues/6070 Add Name Name.unqualifiedName()
- There will be more

Quite possibly, we can add additional convenience similar to what you're requesting. I'm thinking of a `Name.concat(Name)` method.

What's the use-case of appending several "parts" at the end of a `Name` whose qualifier is a table? Are you using Oracle `OBJECT TYPE` or PostgreSQL `TYPE` for nested records?

**Comment by rtomsick** [comment]
Well my real motivation was to get away from having queries littered with

```java
field(name(SOME_TABLE.getName(), "bar"))
```
since the repeated `getName()` calls add a bunch of visual clutter.

Actually the use case that prompted me to make the request didn't need multiple parts; I just figured that it would be a reasonable thing to allow.  Not being intimately familiar with all of the DB types that jOOQ supports I figured there was a reason for the variadic form of `#name()`, thus it should be preserved even with the convenience method.

**Comment by lukaseder** [comment]
I see - yes that kind of feature has been requested before. I cannot seem to find the reference, but I can certainly see how that would be useful

**Comment by lukaseder** [comment]
I'll think about this more thoroughly. Probably, there are quite a few other types that would profit from this, not just the `Table`/`String` (fieldName) combination: https://github.com/jOOQ/jOOQ/issues/6265

**Comment by lukaseder** [comment]
I think the appropriate way forward here is to extract API from various types to a new `org.jooq.Named` type (see #6265), which will simplify interacting with names.

I think that with the features implemented in jOOQ 3.10 (see https://github.com/jOOQ/jOOQ/issues/6089#issuecomment-294531929) and the new `Named` type, we'll have your requirements covered. So, I'm closing this as a dupliate of the other issues.
